common/djangoapps/student/migrations/0011_course_key_field_to_foreign_key.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import django.db.models.deletion
+import openedx.core.djangoapps.xmodule_django.models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('course_overviews', '0013_courseoverview_language'),
+        ('student', '0010_auto_20170207_0458'),
+    ]
+
+    operations = [
+        # Pin the db_columns to the names already in the database
+        migrations.AlterField(
+            model_name='courseenrollment',
+            name='course_id',
+            field=openedx.core.djangoapps.xmodule_django.models.CourseKeyField(max_length=255, db_index=True, db_column='course_id'),
+        ),
+        migrations.AlterField(
+            model_name='historicalcourseenrollment',
+            name='course_id',
+            field=openedx.core.djangoapps.xmodule_django.models.CourseKeyField(max_length=255, db_index=True, db_column='course_id'),
+        ),
+
+        # Rename the fields in Django to the new names that we want them to have
+        migrations.RenameField(
+            model_name='courseenrollment',
+            old_name='course_id',
+            new_name='course',
+        ),
+        migrations.RenameField(
+            model_name='historicalcourseenrollment',
+            old_name='course_id',
+            new_name='course',
+        ),
+
+        # Alter the fields to make them ForeignKeys (leaving off the db_constraint so
+        # that we don't create it at migration time). The db_column is left off because
+        # it defaults to ${field_name}_id, which we pinned it to up above.
+        migrations.AlterField(
+            model_name='courseenrollment',
+            name='course',
+            field=models.ForeignKey(db_constraint=False, to='course_overviews.CourseOverview'),
```

Comment:
The django app that generates the history model automatically makes the historical table `DO_NOTHING`, so that if you did have a cascade on the main table, it wouldn't delete the history.